ludwig/schema/utils.py

```diff
@@ -577,6 +578,31 @@ def IntegerRange(
     )
 
 
+@DeveloperAPI
+def Float(default: Union[None, float], allow_none=False, description="", parameter_metadata: ParameterMetadata = None):
```

Comment:
How is this meaningfully different from `FloatRange`? The strictness doesn't [add too much for floats atm](https://github.com/marshmallow-code/marshmallow/issues/1601) - are you seeing a specific use case for it in hyperopt?

Reply:
This is mostly an issue of semantics, and can absolutely be removed if I'm overthinking it. There were a couple of considerations that led to `Float`:

- It's difficult to set generalized ranges for hyperparameter values. Different params will have different viable ranges, so for a general parameter config object, the range would need to be `[-inf, inf]`.
- `lower`/`upper` and `mean`/`sd` act as bounds on the viable range for a given parameter anyway, and each is represented by a single float/int

My thinking was that a singular unbounded float represented the intent of the fields better than `FloatRange`, but it may also be overkill to add an entire field type just for hyperopt.

Reply:
Gotcha. By default `FloatRange` is a singular unbounded float (leaving the `max` and `min` arguments set to `None` is conceptually the same as `[-inf, inf]`)

Reply:
Should I remove `Float` and replace those fields with `FloatRange`?

Reply:
You could keep it around as a simplified alias that uses `FloatRange` under the hood, but we shouldn't have two separate, duplicated implementations

Reply:
Followup: removed Float and converted those instances to FloatRange.